core/src/main/java/org/apache/iceberg/avro/ValueReaders.java

```diff
@@ -586,37 +588,51 @@ protected StructReader(List<ValueReader<?>> readers, Schema schema) {
       }
 
       if (isDeletedColumnPos == null) {
-        this.positions = new int[0];
-        this.constants = new Object[0];
+        this.constantValuesPositions = new int[0];
```

Comment:
"ValuesPositions" is a bit awkward with multiple plural parts. I think it is also longer than needed because the variables would make sense as `constantPositions` and `constantValues`.

Reply:
Used `constantPositions`, `constantValues`, `defaultPositions`, `defaultValues` for consistency.